### PR TITLE
Bump version to 3.0.0rc1

### PR DESCRIPTION
**Pull request by hhaAndroid**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Bump version to 3.0.0rc1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,13 +75,11 @@ Apart from MMDetection, we also released [MMEngine](https://github.com/open-mmla
...
-**v3.0.0rc0** was released in 31/8/2022:
+**v3.0.0rc1** was released in 26/9/2022:
 
-- Unifies interfaces of all components based on [MMEngine](https://github.com/open-mmlab/mmengine).
-- Faster training and testing speed with complete support of mixed precision training.
-- Refactored and more flexible [architecture](https://mmdetection.readthedocs.io/en/3.x/overview.html).
-- Provides more strong baselines and a general semi-supervised object detection framework. See [tutorial of semi-supervised detection](https://mmdetection.readthedocs.io/en/3.x/user_guides/semi_det.html).
-- Allows any kind of single-stage model as an RPN in a two-stage model. See [tutorial](https://mmdetection.readthedocs.io/en/3.x/user_guides/single_stage_as_rpn.html).
+- Release [RTMDet](configs/rtmdet).
+- Refactored anchor head and base head with `boxlist`.
```
Comment (RangiLyu) [inline README.md:81]: boxlist is not the official name. May use "box type".

Comment (hhaAndroid) [reply README.md:81]: OK

File: README.md [hunk 2]
```diff
@@ -75,13 +75,11 @@ Apart from MMDetection, we also released [MMEngine](https://github.com/open-mmla
...
+- Refactored `SemiBaseDetector` and `SoftTeacher`.
```
Comment (ZwwWayne) [inline README.md:82]: We can simply leave lines 80-81 and skip this.

Comment (hhaAndroid) [reply README.md:82]: delete L82?

Comment (ZwwWayne) [reply README.md:82]: yes

**Pull request merged**